Draft for TF meeting best practices (#39)
swicg/meetings: MEETING-BEST-PRACTICES.md

## Best practices for SWICG-related meetings

### Task forces

#### Before the meeting

> * The task force chair picks a meeting time.

Reviewer: ```suggestion
* The task force lead picks a meeting time.
```

> * The task force chair sets up an online meeting space for that time. Many SWICG people prefer https://meet.jit.si.

Reviewer: ```suggestion
* The task force lead sets up an online meeting space for that time. Many SWICG people prefer https://meet.jit.si. This is a useful tool for working out timezones: https://time.philna.sh/
```

Author: Lots of tools out there including the functionality in the W3C calendar. Leaving this out.

Reviewer: just do the chair->lead change then. This is really just a nice "let's figure out timezones tool. This is another good one for coordinating multiple people: https://timeful.app/ (no ads)

  There isn't really a room to reuse, so pick a room that works for the event.

> * The task force chair creates an issue in the Github repo of the task force with the draft agenda

Reviewer: ```suggestion
* The task force lead creates an issue in the Github repo of the task force with the draft agenda
```

  for the meeting. (Example: https://github.com/swicg/activitypub-e2ee/issues/82)

> * The task force chair notifies one of the SWICG chairs that the meeting will take place with

Reviewer: ```suggestion
* The task force lead notifies one of the Social Web CG chairs that the meeting will take place with:
```

Author: Given this is under github.com/swicg I think we can stick with the acronym, people will know what it is.

Reviewer: Sure, I guess, but technically `swicg` is outdated, we're no longer swicg, but swcg, we're not incubating anymore.

  * Date, time, timezone, expected duration
  * Location (e.g. Jitsi URL)
  * Meeting title (e.g. "Meeting of the xyz Task Force")

>   * Link to the Github issue that contains the agenda.

Reviewer: You don't want this, since it can't be changed in the w3c calendar for each meeting. Perhaps specify if it's a one-off or recurring meeting. One-off can have agenda, but recurring can't.

Reviewer 2: Idunno, you could have a running agendaS issue, seems fine?  I'd recommend if you do that making the first comment on the issue the long-term "status tracker" for epics or parking lot/backlog for running/future topics, and just add a new comment each meeting for the next agenda.  this creates a lot of github "edits", which can be a bit of nightmare to audit, so better for informative things like agendas (and terrible for DECISIONS, which should not be in iteratively-edited github comments!).

* The SWICG chair puts the meeting on the W3C calendar.

> * The task force chair sends the same info to the SWICG mailing list. Invite people to comment on the Github

Reviewer: ```suggestion
* The task force lead sends the same info to the SWICG mailing list. Invite people to comment on the Github
```

  issue with the agenda if they have additions/suggestions.

#### During the meeting

* The task force chair makes sure an online document has been set up for note taking. The document should be
  editable by multiple people. Some services that have been used:

  * https://pad.w3.org -- hosted by the W3C, requires W3C login
  * HedgeDoc

>   * CryptPad

Reviewer: For meeting notes, there's https://notes.socialcg.org/ — this is a service I operate for the Social Web CG (and WG if they want), if you need an account, please request one, and give me 24-48 hours to set it up. Details: https://notes.socialcg.org/about

Full backstory: https://activitypub.space/topic/80/taking-notes-for-social-web-cg-taskforce-meetings?_=1779918891694

* The task for chair shares that document in the meeting chat and identifies one or more people who will help with

>   taking notes.

Reviewer: If no one can take notes, then the meeting should be recorded and transcribed; this is the only way to capture accurate notes in small meetings.

Note: recording meetings only appears to work in chrome with jitsi.

I use this prompt to generate a condensed version of the notes, since direct conversation captured via transcription can often be messy. 

You can either use an app to transcribe first or just directly supply the audio recording:

Prompt: https://gist.github.com/ThisIsMissEm/35559c2110356b4d38fdf236b2e31a5a

Reviewer: Obviously using AI is controversial and optional, but I strongly recommend it: AI transcription & meeting notes are preferable to no meeting notes at all, or very condensed meeting notes that cannot easily be understood by a reader who wasn't present at the meeting.

> * The task force chair reminds attendees of the W3C Code of Conduct, and W3C IP policy.

Reviewer: I would recommend setting up an issue template, which provides you the markdown with all the administriva: https://github.com/swicg/activitypub-trust-and-safety/blob/main/.github/ISSUE_TEMPLATE/meeting-template.md

Author: I don't want to clutter this document with recommendations how to use Github.

Reviewer: It's more just a tip: "You can use an issue template for saving time here!"

#### After the meeting

* The task force chair turns the meeting notes into a pull request for the SWICG meeting repo at:

>   https://github.com/swicg/meetings

Reviewer: As far as I know you can either use that folder or meetings/ within your own github repository. The latter is slightly easier to manage as you have control over your own repository and don't need approval.

Reviewer: ```suggestion
* The task force lead turns the meeting notes into a pull request for the SWICG meeting repo at: https://github.com/swicg/meetings
* OR, the task force lead turns the meeting notes into a pull request for their own repository, using the `meetings/YYYY-MM-DD-taskforce-meeting.md` naming scheme.
```
